### DISCUSSION TITLE
Decentralized storage on personal data pods by leveraging the Solid ecosystem

### DISCUSSION THREAD

**Discussion by gertjandemulder** [ActivityWatch/activitywatch#838] [Ideas]
# Solid integration

Keywords: Solid, Linked Data, RDF, Decentralization

## Idea

- **Objective**: Store activity data, as linked data, on a Solid pod.
- **Outcome**: This allows users to sync their activity data between multiple machines, without the need for a dedicated server

## Solid

Solid is a specification that lets people store their data securely in decentralized data stores called Pods. Pods are like secure personal web servers for your data.

- Any kind of information can be stored in a Solid Pod.
- You control access to the data in your Pod. You decide what data to share and with whom (be it individuals, organizations, and/or applications). Furthermore, you can revoke access at any time.
- To store and access data in your Pod, applications use standard, open, and interoperable data formats and protocols.
- You can get a Solid pod from a [Pod Provider](https://solidproject.org/users/get-a-pod), or you may choose to self-host your Pod.

Solid pods can be used to create a Personal Information Management System (PIMS), giving individuals more control over their personal data. Individuals can manage their personal data in secure online decentralized storage systems called pods: secure personal web servers for data. All data in a pod is accessible via the Solid Protocol.

When data is stored in someone’s pod, they control who and what can access it. and share them when and with whom they choose using Solid pods. Providers of online services and advertisers need to interact with the Solid pod if they plan to process the individual’s data. This enables a human-centric approach to personal information and new business models.

Any kind of data can be stored in a Solid pod, however, the ability to store Linked Data is what makes Solid special. Linked Data gives Solid a common way to describe things and how they relate to each other, in a way that other people and machines can understand. This means that the data stored by Solid is portable and completely interoperable.

Anyone or anything that accesses data in a Solid pod uses a unique ID, authenticated by a decentralized extension of OpenID Connect. Solid’s access control system uses these IDs to determine whether a person or application has access to a resource in a pod.

# References

- <https://ruben.verborgh.org/blog/2022/12/30/lets-talk-about-pods/>
- <https://ruben.verborgh.org/blog/2017/12/20/paradigm-shifts-for-the-decentralized-web/>
- <https://github.com/solid/solid-spec>
- <https://solidproject.org/about>
- <https://solidproject.org/users/get-a-pod>
- <https://github.com/inrupt/solid-client-authn-js>

**Comment by ErikBjare** [comment]
I'm slightly familiar with Solid, but what does it have to do with ActivityWatch?

The sync solution we're working on uses a simpler file-system model where users "bring their own sync" (Syncthing, Dropbox, Gdrive) to sync the data. I don't think Solid offers a tangible benefit over that solution, and probably brings additional limitations.

**Reply by xeruf** [reply]
All identity data hopefully in one place in the future, ability to easily share something?
